Inventory, new equips types, equip bonus

azeveco opened the pull request:
Hi guys!
I'm playing this korean version: http://burgerburger.kr
You can play with google translater.

So, there you will find inventory system, skills, new equipments slots (necklace, ring and boots), equipment bonus (like x% drain hp, +5% damage, +2 Defense...), a coin system (they use burgers to buy stuff. There is a public market where players sell items for burgers), quests, lots lots and lots of npc, mobs, items, equips, quests, areas, maps, everything.

So,maybe you guys can do any of this? And please, a system that shows player level on character, like this:
 Lv. 10 - MrMota
      (  º _ º  )
      //l       l\
        l       l
       //       \

Here's a little database of where drops what on this game:
http://www.burgerburger.kr/strategy/246915#1

You guys really should take a look. Lots of stuff
Some screens:
http://www.burgerburger.kr/screenshot/242962
http://www.burgerburger.kr/screenshot/243923

There is much more, just search it.

sork committed:
Offload some traffic to Mozilla CDN and music server. Disable createSilouhette if no CORS headers.

tfausak committed:
Add icons and startup images for iOS devices.

Commit:
Some minor quibbles.

sork committed:
Add mozilla CDN urls

sork committed:
Add crossorigin property

sgricci committed:
Key Movements

sgricci committed:
Typo in 'goUp' logic

bnoordhuis committed:
build: make build.sh independent of $PWD

Users no longer need to `cd /path/to/BrowserQuest/bin && ./build.sh`,
now `/path/to/BrowserQuest/bin/build.sh` is sufficient.

rob-mccann committed:
Added radix of 10 to calls to parseInt

sgricci committed:
Added several more hot keys

sgricci committed:
Fixed an issue where moving by keys would cause screen not to switch

sgricci committed:
Walking in the direction of an NPC will act as if you clicked on them

sgricci committed:
Removing swp files

sgricci committed:
Fixed an issue which would cause you to not be able to move with keys if you were hovering over a colliding tile

bahamas10 committed:
Cleaning up build.sh

* Quoting all variables to avoid possible word splitting

kornelski committed:
Optimized images

sgricci committed:
Added fux for chatbox + keyboard commands

mathiasbynens committed:
Automatically focus the “player name” input field when loading the page

sgricci committed:
Added support for keypad 2,4,6,8 as directional keys

frogstarr78 committed:
Updating code to remove deprecation warning when using path.exists and path.existsSync.

KorvinSzanto committed:
Add link to browser quest, I had to google it =[

glecollinet committed:
Add legal scroll view

glecollinet committed:
Display privacy link on mobile version

glecollinet committed:
Add tooltip above name input

glecollinet committed:
Add placeholder text in chat

glecollinet committed:
Change create a new character > reset your character

glecollinet committed:
Add privacy paragraph + legal notices link

glecollinet committed:
Change copy 'chats' > 'equipment'

Commit:
Added working setup instructions to front README.

Commit:
Wind back the CDN urls, but keep the useful looking code.

Commit:
Added missing background music, improved attribution.

Commit:
No need to git ignore the music files any more.

Commit:
Added @mkrecny, @bnoordhuis and @justinclift to the credits.

Commit:
Added @tfausak, @willurd and @sgricci to the credits.

Commit:
Updated GitHub source code links to the new development repo.

Commit:
Added @bahamas10 to the credits.

Commit:
Added @mathiasbynens to the credits.

Commit:
Added @unforeseen to the credits.

Commit:
Added @frogstarr78 to the credits.

Also fixed the server output example, which the commit by @frogstarr78 addresses.

Commit:
Removed old info about fs.exists warning.

Commit:
Added @pornel to the credits.

Commit:
Fixed Kornel Lesiński's name.

Commit:
Added @KorvinSzanto to the credits.

Commit:
Added v0.8.8 version to Node.js line.

Just in case the Node.js version is causing the problem reported here:

https://github.com/browserquest/BrowserQuest/commit/e94d70a0348c609b6d71ebab635362379c9f3994#commitcomment-1861896

Commit:
Reverting e94d70a for now, as it was breaking on older Node.js.

It will need to made conditional or something, later on.

Commit:
Added path.exists back into the startup example.

Commit:
Added note that Node.js 0.6.x series should work.

Commit:
Added easy way to start a client server.

Importantly, it's both dead simple and lets us
easily spot missing files (ie config files,
shared directory).

Commit:
Trivial typo fix.

Commit:
Merge pull request #2 from justinclift/clientstart

Added easy way to start a client server.

Commit:
Added link to the new mailing list and archives.

Commit:
Simplified the node dependency installation line.

Commit:
Updated server page instructions to point at the root dir.

willurd committed:
Added the 'file' module which abstracts the exists and existsSync functions to support node versions before and after 0.7.1

willurd committed:
Turned the local config warning into a local config notice

Commit:
Added initial client side setup instructions.

Commit:
Fixing brain-o.

Commit:
Added Jeff Lang to the credits.

Commit:
Removed first person reference from front README.

dylenbrandon committed:
Added IRC Channel to README.md

dylenbrandon committed:
Update README.md

dylenbrandon committed:
Update README.md

Missed the hash mark for the irc channel. Looks better now.

dylenbrandon committed:
Extra ../ in requirejs line caused build.sh to fail.

dylenbrandon committed:
Merge branch 'master' of https://github.com/browserquest/BrowserQuest

Commit:
Added Dylen Rivera to the credits.

Commit:
Added wiki link to the front page README.

Commit:
Both client and server can now run over the same port.

Turned on/off by the 'use_one_port' boolean in server/config.json.

After starting the BQ server, connect to the port it's listening
on with your browser.

  i.e.  http://server.example.org:8000/

The BQ server is now serving the client files through the
same port, plus transparently taking care of the client
shared/* and config/* files. (no need to touch them any more
when running over one port)

This approach will probably suit developers and casual
users very well, but may get in the way of multi-node
setups. Change the 'use_one_port' setting in server/config.json
to false to use the previous behaviour.

Commit:
Merge pull request #12 from justinclift/oneport

Initial code to run both client and server over the same port.

Commit:
Added Brynn Bateman to the credits.

Commit:
Added initial Browser Support info.

Commit:
Updated setup instructions for use_one_port.

Commit:
Point people to the README.md in the repo root.

Commit:
Updated to follow our code conventions.

Commit:
Moved start_dev_client.js to the bin subdirectory.

Commit:
Added the music authors to the front page credits.

Commit:
Trivial typo fix in comment.

Commit:
Fix blocker bug in v1.2.

dylenbrandon committed:
Changed always use the dispatcher in production to false in client/js/game.js

I find that setting this to false works as expected while true causes "Connecting to server" problems

More information here: https://github.com/mozilla/BrowserQuest/issues/95

Commit:
Merge pull request #18 from justinclift/v122

Fixes EC2 breaking bug, by now using incoming http req header info.
Thanks to @sgricci for code sanity check. :)

Commit:
Removed trailing whitespace and embedded tabs from Node.js source.

Commit:
Merge pull request #20 from justinclift/nodewhitespace

No code changes.

Commit:
Client js: tabs->spaces, nuke trailing whitespace.

Done using:

  $ for i in `ls -1 *js`; do sed -i '' 's/[  ]/    /g' $i; done

  The character in the [ ] is a tab character, not a space.

then:

  $ for i in `ls -1 *js`; do sed -i '' 's/[  ]*$//' $i; done

  The characters between [  ] are 1 x space character, then 1
  x tab character.

In bash, the tab character can be typed using Control-v then
Control-i.

Commit:
Merge pull request #21 from justinclift/clientwhitespace

No code changes.

Commit:
Background music doesn't play in Safari 6.x.

Commit:
Removed trailing whitespace.

Commit:
First attempt with using jshint.

Unsure about the module.exports bit.  Seems to work though.

Commit:
Force older underscore module for now.

Underscore 1.4.0 is causing TypeError exceptions:

  ERROR uncaughtException: TypeError: Cannot read property 'forEach' of undefined

Commit:
Ignore .project files, for Eclipse based editors.

Commit:
Updated checkpoint.js for JSHint.

Commit:
Updated chest.js for JSHint.

Commit:
Updated chestarea.js for JSHint.

Commit:
Updated entity.js for JSHint.

Commit:
Added character.js for entity cleanup.

Probably conflicts with the proper character.js cleanup in the
other branch though.

Commit:
Further refining of area.js.

Commit:
Adjusted utils.js for JSHint.

Commit:
Adjusted npc.js for JSHint.

Commit:
Adjusted ws.js for JSHint.  Still needs work.

JSHint still reports many errors, however its a lot better than
it was and everything still seems to work.

Commit:
Removed the indent:4 JSHint option.

It causes bogus error messages with switch statements.  Seems like
a bug in the version of JSHint being used.

willurd committed:
Merge pull request #22 from justinclift/oldunderscore

Force older underscore module for now.

willurd committed:
Merge pull request #23 from justinclift/jshintarea

First attempt with using jshint

Commit:
Make uncaughtException errors log useful info.

Commit:
Disable strict mode, as its causing problems.

The 'problems' might really be with our codebase, but
investigating that is a task for later.

Commit:
Fixes an error about ChestArea not being defined.

Commit:
Updated character.js for JSHint.

Commit:
Updated format.js for JSHint.

Commit:
Removed indent option, as it breaks on switch statements.

Commit:
Updated formulas.js for JSHint.

Commit:
Updated item.js for JSHint.

Commit:
Added better debugging output for ws.js.

Commit:
Started adjusting maps.js for JSHint.

Seems insanely sensitive to otherwise straightforward changes.

Commit:
Save point.

Commit:
Save point.

Commit:
Removed whitespace from shared js files.

Commit:
Added further debugging detail.

Commit:
Updating gametypes to be more resilient.

Commit:
Added log parameter to some functions.

Commit:
Updated map.js for JSHint.

Commit:
Updated message.js for JSHint.

Commit:
Updated memcache metrics.js for JSHint.

Commit:
Moved log setup to the start, so other functions can use it.

Commit:
Now passes the log to the metrics include.

Commit:
Updated to be smarter with the log variable.

Commit:
First pass cleanup for mob.js.

Commit:
Updated mob.js for JSHint.

Commit:
Updated mobarea.js for JSHint.

Commit:
Enabled random roaming for the monsters.

Commit:
Moved jshint options into .jshintrc file.

Commit:
Fixed weird freezing issue when walking over items.

Seems to be related to the underscore and embedded class modules.

Commit:
Exclude the WebStorm IDE project files dir.

Commit:
Move the collision grid log entries to debug level.

Commit:
Stop passing the log variable to each module.

Was completely unnecessary, as by the time any module's functions
are called it's already available to them (from callers scope).

Commit:
Gave anonymous functions decent names.

For better stack traces.

willurd committed:
Merge pull request #24 from justinclift/jshintnode3

Next bunch of JSHint and general cleanup changes.

Commit:
Conditional NPC talk & entry point

Edit of npcs (see scientist for an example) :
Discourse can now be circonstancial. One can specify a condition under the form of a parameterless function for each sequence of sentences. It should still be working for existing npc, but others can adapt their talk.
 * The form, which is not tested is an array of objects. Each object (except the first, default discussion) has a "condition" property (parameterless function) which decides when to use this discussion and a "text" property, which contain the array of lines of the onversation.
 * The variable made available to the functions is the game itself.

Added "entry point" :
An entry point is used to get into the game through encoded urls (used sha1 by caligatio http://caligatio.github.com/jsSHA) to identify them. This can be used to establish connexion with the outside world, might it be on the Web (achievement obtained through activity on another site, section of the site) or in the real world (QRCode found on an item).
Can be tested with

* Corrected hasty commit (changed place where the Entry Point is called)
* Added an example : ./client/?entrance=almostFoxy
* hastily defined *name* as the name of the user in npc dialogs

Almost forgot to reinstate rick…

Commit:
Merge pull request #26 from lzbk/master

Tested and working on Node.js v0.8.18 on Win 8.

Commit:
Added Mathieu Loiseau to the credits.

Commit:
Added Windows 8 info.

Mawgamoth committed:
Fixed and smoothed out keyboard movement.  Fixed a rare bug regarding talking to NPCs.

Commit:
Merge pull request #29 from Mawgamoth/master

Keyboard Movement Changes

Commit:
Added Jason Culwell to the credits.

Commit:
Added OSX install instructions

Mawgamoth committed:
Corrected issue with keyboard movement and talking NPCs.  Made cursor hide when using keyboard movement.

Commit:
added guilds (and debug messages)

gatekeep committed:
Rewrote processmap.js to use JSON files outputted by latest Tiled.

Commit:
Merge pull request #30 from Mawgamoth/master

Fixed issue with keyboard movement and talking NPCs.  Made cursor hide when using keyboard movement.

Commit:
few bug corrections + removed debug notifications

Commit:
Merge pull request #33 from gatekeep/master

Updated the processmap.js script to use the output from latest Tiled

Commit:
Merge pull request #34 from lzbk/master

Guilds

Commit:
Added Bryan Biedenkapp to the credits.

Sorry for forgetting Bryan, my bad.

Commit:
Map export tool updated and documented getting rid of python

Commit:
Updated and documented map export tool, getting rid definitively rid of python.

Commit:
Merge branch 'master' of https://github.com/lzbk/BrowserQuest

Commit:
Merge pull request #40 from lzbk/master

Facilitating and documenting map export (no Python)

Commit:
Update README.md

typo

Commit:
Merge branch 'master' of git://github.com/browserquest/BrowserQuest

Commit:
more map utilities

Commit:
Merge pull request #41 from lzbk/master

a quick and dirty script to facilitate extending the tilesheet

Commit:
Updated the versions of Node.js known to work

Commit:
Improved formatting of OSX instructions

Commit:
Improve OSX instructions, update browser support list

Commit:
Trival tweak to browser support list

Aaron1011 committed:
Made player chase selected target instead of stopping on selected square

Adjusted indentation to 4 spaces per level

Commit:
Typo fix

Aaron1011 committed:
Added Aaron1011 to contributers list

fredriksvantes committed:
Updated so pvp, database backend saves, target hovering, leveling system is available. Also changed cli for guilds ( /guild invite, /guild create etc)

Commit:
Added OSX Homebrew install instructions for Redis + other minor fixups

Commit:
Pedantic typo fix

Missed this one, should have been in the last commit :)

Aaron1011 committed:
Moved redis server information into config file

Commit:
Fixed a typo in the RHEL/CentOS/Fedora redis service start command

Aaron1011 committed:
Fix server metrics

Added necessary configuration variables for metrics to work

Fixed properly exporting Metrics class

Store world count in Memcache

Aaron1011 committed:
Pick up items when walked over

Aaron1011 committed:
Added background to enemy stats hover window

Commit:
Improved instructions for Fedora/CentOS/SL

Aaron1011 committed:
Added symbolic link to server/js/main.js in root directory

Aaron1011 committed:
Setup Memcache on Openshift

Aaron1011 committed:
Setup connection to Redis cartridge

Aaron1011 committed:
Made scripts executeable

Aaron1011 committed:
Made server use Openshift IP and port

Aaron1011 committed:
Setup installation of custom version of NodeJS

Aaron1011 committed:
Fixed indentation in server/js/databasehandler.js

Aaron1011 committed:
Added attribution to external scripts

Aaron1011 committed:
Added instructions for deploying to Openshift

Aaron1011 committed:
Allow configuration of different production hosts

Commit:
Match the OpenShift Redis env var names to what the cartridge provides

Commit:
Improve the visibility of the Redis cartridge add command

Aaron1011 committed:
Removed extra console logging statement

Aaron1011 committed:
Made enemy stats window not show 'Level undefined' for non-enemies

Aaron1011 committed:
Made enemy stats display only show name for objects and NPCs

Aaron1011 committed:
Made enemy stats window disappear after two seconds when the target is no longer hovered

Aaron1011 committed:
Fixed display problems with enemy stats window

Aaron1011 committed:
Added initial unit tests

Aaron1011 committed:
Added unit tests for client/js/area.js

Aaron1011 committed:
Moved test boilerplate code into a separate file

Aaron1011 committed:
Added Travis CI build status image to README.md

Aaron1011 committed:
Updated URL in README.md to point to new BrowserQuest instance

Aaron1011 committed:
Added Openshift hot deploy marker

Aaron1011 committed:
Moved boilerplate.js from client/ to shared/js

Aaron1011 committed:
Added unit tests for client/js/exceptions.js

Aaron1011 committed:
Added javascript minification

Aaron1011 committed:
Fix Redis environment variables

Aaron1011 committed:
Added support for multiple database backends

Aaron1011 committed:
Added deploy section to .travis.yml

Aaron1011 committed:
Updated .travis.yml for conditional deploy

Aaron1011 committed:
Removed repository line from .travis.yml

Aaron1011 committed:
Revert "Added deploy section to .travis.yml"

The conditional deployment section in .travis.yml caused BrowerQuest to not be
deployed at all. Until I can figure out how to fix it, I've reverted to
the earlier code which deploys on all builds (but still works).

Aaron1011 committed:
Fixed up package.json

Commit:
Merge pull request #84 from Aaron1011/fix_package

Fixed up package.json

Aaron1011 committed:
Added unit test for Item

Aaron1011 committed:
Use static method notation for Area.init

Commit:
Merge pull request #79 from Aaron1011/fix_redis_config

Fix Redis environment variables

Aaron1011 committed:
Deploy to Team BQ instance in .travis.yml

Aaron1011 committed:
Revert "Added javascript minification"

This reverts commit 1f263f39eca93790c2c0f90cb64b451e3fc6acb6.
The minification is causing problems on Openshift.

Conflicts:
	package.json

justinclift commented:
@MrMota - Heh, while I like the idea of the browserquest/BrowserQuest repo being merged into the upstream Mozilla one, it's a very different thing from the text description of this PR (which is about burgerburger.kr's fork).

So, good luck? :smile:

As a secondary thought, do you know if the burgerburger.kr guys have their source repo online somewhere?  That would make it a lot easier to look at their code + copy the useful bits. :smile:

Aaron1011 committed:
Added irc notifications to .travis.yml

Commit:
Updated info about Opera support

Aaron1011 committed:
Added repository field to package.json

Aaron1011 committed:
Changed credentials in .travis.yml

Now that Aaron1011 owns the teambq domain, the deploy section needs to
be updated with his credentials.

Commit:
Updated to use Node 0.8 or 0.10, as 0.6 doesn't work

Aaron1011 committed:
Disable hot deploy

Aaron1011 committed:
Fixed Openshift credentials in .travis.yml

Aaron1011 committed:
Added Gemnasium status image

Aaron1011 committed:
Updated underscore dependency

Commit:
Merge pull request #110 from Aaron1011/update_underscore

Updated underscore dependency

Aaron1011 committed:
Added 'Fork me on Github' ribbon to home page

Commit:
Updated with the new mailing list details

Aaron1011 committed:
Merge pull request #114 from justinclift/readme_add_new_mailing_list

Updated with the new mailing list details [ci skip]

Commit:
Updated link to public mailing list archives

lukas-shawford committed:
Fix issue with not being able to enter 'a' or 'f' in login form, and other minor fixes

These were apparently used for debugging purposes ('a' ran the attack animation, and 'f' shows the FPS). The actual functionality was commented out, but the 'return false' part was left in there, which prevented the default action from bubbling up properly.

Explanation for some of the other minor fixes:

- remove app.hideDropDialog(): This is not actually defined. More importantly, it's causing the other ESC key handler in this file from firing (search for the last occurrence of app.hideWindows), which closes any open dialogs (such as achievements or help).

- The event handler for 'a' that called game.player.hit (note: the call was commented out because it was for debugging purposes) was removed entirely since it conflicted with the WASD movement keys (so even if you were to uncomment it, it still wouldn't do anything). Besides, all it did was run the player's attack animation without actually attacking anything, even if you were standing next to someone - probably not terribly useful.

Aaron1011 committed:
Merge pull request #122 from sergkr/master

Fix issue #118 ('a' and 'f' keys do not work in login form)

Commit:
Added Sergey Krilov to the credits

lukas-shawford committed:
Allow using the Enter key to log in (fixes issue #104)

lukas-shawford committed:
Add some basic validation to the create new character / login forms (fixes issue #116)

Aaron1011 committed:
Merge pull request #123 from sergkr/issue104

Fix for issue 104 (allow Enter key to log in), and maybe a partial fix for 116 (warn about mismatched passwords)

nickdesaulniers commented:
Thanks for taking a look at the source code for BrowserQuest!  BrowserQuest was developed by [Little Workshop](http://www.littleworkshop.fr/) on behalf of Mozilla to showcase how the Web is the ultimate cross platform for games.

Mozilla does not currently have the resources to maintain BrowserQuest, but you should check out some of these amazing community forks instead!  Feel free to add your fork to the [wiki](https://github.com/mozilla/BrowserQuest/wiki/Community-Forks).

**English**
- [browserquest/browserquest](https://github.com/browserquest/BrowserQuest) - The general development group
  - Running game here, you're welcome to join: http://browserquest-teambq.rhcloud.com:8000
- [ParticleQuest](https://github.com/particlequest/ParticleQuest) - Making particle physics fun. The goal of this project is to create a science based story modifying BrowserQuest.

**Korean**
- Burger Burger Online: http://burgerburger.kr - Extremely well developed, significantly expanded from the original BrowserQuest
  - Community: http://asky.freethinker.kr
  - English is being added
  - Older source: https://github.com/browserquest/BrowserQuest-Asky

**Android**
- [BrowserQuest for Android](https://svn.strasweb.fr/listing.php?repname=Pierre+Rudloff&path=%2Fbrowserquest%2F) - BrowserQuest client, [available in the Android store](https://play.google.com/store/apps/details?id=fr.strasweb.browserquest)

**Windows Phone**
- [BrowserQuest for Windows Phone](http://www.windowsphone.com/en-US/store/app/browserquest/9df4ba1e-fa73-40d1-b391-4773d287db35) - BrowserQuest, in Windows Phone store

**Erlang based**
- [sedrik/bqs](https://github.com/sedrik/bqs) - Erlang implementation of the BrowserQuest server

**C++ based**
- [chain710/browserquestcc](https://github.com/chain710/browserquestcc) - C++ implementation of the BrowserQuest server

## Related
- [HabitRPG](https://habitrpg.com/static/front) - A habit building program which treats your life like a Role Playing Game
  - Uses the BrowserQuest character icons (CC-licensed) for some of their stuff, and is looking to contribute back the icons they've developed since

## No longer actively developed

**French**
- UFB World: http://ufbteam.com/fr
  - Source: https://github.com/hackziner/ufbworld
